[Ruihan] IP

ruihan00 opened the pull request:
# Duke X Amogus
> I was at reactor.

Duke's newest collaboration with Amogus frees your mind of having to remember things you need to do. Its

> BLACK SELF REPORT!
- Text based
- Easy to learn
- SUPER FAST TO USE

> I saw red vent.

All you need to do is, 
1. download it from [here](https://github.com/ruihan00/ip)
2. double-click it
3. add your tasks
4. let it manage your task for you!!

And it is **FREE**!

> White sus.

Features:
- [x] Manage tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

ruihan00 committed:
Level-1

ruihan00 committed:
Level-2

ruihan00 committed:
Level-3

ruihan00 committed:
Level-4

ruihan00 committed:
A-TextUiTesting

ruihan00 committed:
Level-5

ruihan00 committed:
Level-6

ruihan00 committed:
A-Enums

ruihan00 committed:
A-TextUiTesting

ruihan00 committed:
A-TextUiTesting

ruihan00 committed:
test pr

ruihan00 committed:
Level-7

ruihan00 committed:
Level-8

ruihan00 committed:
A-MoreOOP

ruihan00 committed:
move into duke package

ruihan00 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'
Add gradle branch

ruihan00 committed:
edit build.gradle

ruihan00 committed:
update OOP

ruihan00 committed:
Add Junit Test

ruihan00 committed:
add javaDoc

ruihan00 committed:
change case indentation

ruihan00 committed:
add find function

JustinS00 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,53 @@
+package duke;
+import java.time.LocalDate;
```
Maybe add a line in between the package and imports

JustinS00 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,80 @@
+    public void showGreeting() {
+        System.out.println("Howwwwwdyyyyyy");
+        this.lineBreak();
+        String image = "⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⣠⣴⣶⣿⣿⣷⣶⣄⣀⣀⠀⠀⠀⠀⠀⠀⠀⠀\n⠀"
```
pretty cool

tyw2811 replied:
Agreed!

JustinS00 reviewed (commented):
LGTM! cool stuff

tyw2811 commented on src/main/java/duke/Exceptions/descriptionException.java:
```diff
@@ -0,0 +1,11 @@
+package duke.Exceptions;
+public class descriptionException extends Exception {
```
Perhaps you could add a line between these two lines. I noticed this issue in other places as well.

tyw2811 reviewed (commented):
LGTM! Just a few nits, like missing JavaDocs.

ruihan00 committed:
A-Checkstyle

ruihan00 committed:
Level-10

ruihan00 committed:
Level-10

ruihan00 committed:
Merge branch 'branch-Level-10'
Merge branch

albertZhangTJ commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,80 @@
+    /**
+     * Enum list of commands
+     */
+    public enum Commands {
```
Nice job using enum to differentiate commands.

albertZhangTJ commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,80 @@
+     * Runs the Duke program
+     */
+    public void run() {
+        Scanner scanner = new Scanner((System.in));
```
I don't think a double bracket is needed. Probably a typo?

albertZhangTJ commented on src/main/java/duke/Exceptions/NoSuchCommandException.java:
```diff
@@ -0,0 +1,12 @@
+package duke.Exceptions;
+
+public class NoSuchCommandException extends Exception{
```
I really like how you customize your own exception for this!

albertZhangTJ commented on src/main/java/duke/Task/Event.java:
```diff
@@ -0,0 +1,27 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Event extends Task {
```
My understanding is that `Event` should be a task with two time points (one for start and one for end). It seems that the current definition is somewhat duplicate with the `Deadline` class.

albertZhangTJ commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,165 @@
+                handleDeleteCommand(ui, storage, taskList, inputArr[1]);
+                break;
+            case FIND:
+                handleFindCommand(ui, taskList, inputArr[1]);
```
I would suggest adding a default here

ruihan00 committed:
Level 10

ruihan00 committed:
Merge branch 'branch-Level-10'
merge

ruihan00 committed:
bug

ruihan00 committed:
Merge branch 'branch-Level-10'
merge level10

ruihan00 committed:
add assertions

ruihan00 committed:
A-CodeQuality

ruihan00 committed:
A-CodeQuality

ruihan00 committed:
fixed storage

ruihan00 committed:
remove redundant files

ruihan00 committed:
Merge pull request #2 from ruihan00/branch-A-Assertions

add assertions

ruihan00 committed:
Merge pull request #3 from ruihan00/branch-A-CodeQuality

A-CodeQuality

ruihan00 committed:
add switch file command

ruihan00 committed:
Merge pull request #4 from ruihan00/branch-C-FlexibleDataSource

add switch file command

ruihan00 committed:
Add switch command

Added switch command to enable users to switch data source files on command

Added error checking for file type, files must end with .txt

ruihan00 committed:
Merge branch 'master' of github.com:ruihan00/ip

ruihan00 committed:
add ui png

ruihan00 committed:
add ui png

ruihan00 committed:
add ui png

ruihan00 committed:
add readme

ruihan00 committed:
change filename

ruihan00 committed:
change image size

ruihan00 committed:
add javadocs

ruihan00 committed:
edit readme

ruihan00 committed:
edit readme

ruihan00 committed:
edit readme

ruihan00 committed:
edit readme

ruihan00 committed:
edit readme

ruihan00 committed:
edit readme

ruihan00 committed:
edit readme

ruihan00 committed:
edit readme
